### ISSUE TITLE
pureconfig-magnolia generates invalid ConfigReader for a case class with all private fields

### ISSUE THREAD

**Issue by neko-kai** [pureconfig/pureconfig#702]
Example:

```scala
final case class Abc(
  private val a: Int,
  private val b: Long,
  private val c: String
)
object Abc extends App {
  implicit val reader: pureconfig.ConfigReader[Abc] = pureconfig.module.magnolia.semiauto.reader.deriveReader[Abc]

  pureconfig.ConfigSource.string("{}").loadOrThrow[Abc]
}
```

Expected:
Success. Or at least an error mentioning `a` field

But got:
An error mentioning `a-$access-$0` field
```
Exception in thread "main" pureconfig.error.ConfigReaderException: Cannot convert configuration to a Abc. Failures are:
  at the root:
    - Key not found: 'a-$access-$0'.

	at pureconfig.ConfigSource.loadOrThrow(ConfigSource.scala:80)
	at pureconfig.ConfigSource.loadOrThrow$(ConfigSource.scala:77)
	at pureconfig.ConfigObjectSource.loadOrThrow(ConfigSource.scala:91)
```

**Comment by neko-kai** [comment]
This is fixed in magnolia-0.12.8 - just need to update the dependency in pureconfig-magnolia https://github.com/propensive/magnolia/pull/212

**Comment by ruippeixotog** [comment]
This should be fixed now, as Magnolia was updated quite some time ago (#714). Thanks for updating this issue!
